code/__DATASTRUCTURES/priority_queue.dm

```diff
@@ -0,0 +1,65 @@
+
+//////////////////////
+//PriorityQueue object
+//////////////////////
+
+//an ordered list, using the cmp proc to weight the list elements
+/PriorityQueue
```

Comment:
I'll look into that for a future PR, I'm not fully familiar with Priority Queues personally, just moving their contents to the (now) best place, the datastructure folder.